copier/user_data.py

```diff
@@ -450,13 +445,25 @@ def load_answersfile_data(
         return {}
 
 
-def cast_answer_type(answer: Any, type_fn: Callable) -> Any:
+def cast_answer_type(answer: Any, type_name: str) -> Any:
     """Cast answer to expected type."""
+    try:
+        type_fn = CAST_STR_TO_NATIVE[type_name]
+    except KeyError:
+        raise InvalidTypeError("Invalid answer type")
```

Comment:
I'm not sure how to cover this error by a test as it doesn't occur in practice because the question type is checked to be one of the keys of `CAST_STR_TO_NATIVE` before. Nevertheless, this function could theoretically be called in a different context, so I think it makes sense to keep the check here.